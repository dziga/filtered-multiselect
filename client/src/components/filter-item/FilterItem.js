import React from 'react';
import './FilterItem.scss'

function FilterItem (props) {
    const {item, onItemSelected} = props
    const onClick = () => onItemSelected(item)
    return (
      <div className='item'>
        <label><span dangerouslySetInnerHTML={{__html: item.label}}></span>
          <input type="checkbox" name="checkbox" id={item.id} checked={item.checked} onChange={onClick} />
          <span className='checkmark'></span>
        </label>
        </div>
    );
}

export default FilterItem;
