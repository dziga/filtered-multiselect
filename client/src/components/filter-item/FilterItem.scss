@import '../common/variables';

.item {
  display: block;
  position: relative;
  padding-left: 35px;
  margin-bottom: 12px;
  min-width: 299px;
  cursor: pointer;
  font-size: $font-small;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
    height: 0;
    width: 0;
  }

  /* Create a custom checkbox */
  .checkmark {
    position: absolute;
    top: 0;
    left: 0;
    height: 16px;
    width: 16px;
    background-color: white;
    border: 2px solid $border;
  }
  /* Create the checkmark/indicator (hidden when not checked) */
  .checkmark:after {
    content: "";
    position: absolute;
    display: none;
  }

  /* Show the checkmark when checked */
  input:checked ~ .checkmark:after {
    display: block;
  }

  /* Show the checkmark when checked */
input:checked ~ .checkmark:after {
  display: block;
}

/* Style the checkmark/indicator */
.checkmark:after {
  left: 2px;
  top: 2px;
  width: 12px;
  height: 12px;
  background-color: #3063D0;
}

}