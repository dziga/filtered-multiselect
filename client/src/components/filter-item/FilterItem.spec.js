import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import FilterItem from './FilterItem';

test('filter item should render with given props', () => {
  const item = {id:1, checked:true, label:'book'}
  const component = renderer.create(
    <FilterItem item={item} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('filter item should return item on click', () => {
  const item = {id:1, checked:false, label:'book'}
  const onItemSelected = jest.fn()
  const component = shallow((<FilterItem item={item} onItemSelected={onItemSelected}/>))

  component.find('input').simulate('change', { target: { checked: true } })

  expect(onItemSelected.mock.calls[0][0]).toEqual(item);
})
