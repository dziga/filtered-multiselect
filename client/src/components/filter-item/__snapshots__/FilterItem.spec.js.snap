// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter item should render with given props 1`] = `
<div
  className="item"
>
  <label>
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "book",
        }
      }
    />
    <input
      checked={true}
      id={1}
      name="checkbox"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="checkmark"
    />
  </label>
</div>
`;
